lib/regex.dl

```diff
@@ -11,4 +11,17 @@
  * Returns false if `regex` is not a valid regular expression or if no match was
  * found.
  */
-extern function regex_match(regex: string, str: string): bool
+extern function regex_match(regex: string, text: string): bool
```

Comment:
I wonder whether you may want to have an extern type for regex, which is a compiled regular expression.
If you want to use a regex in a rule it would be nice to be able to compile it.

Reply:
The current API is a convenience API that combines regex compilation and matching. It works nicely for our existing use cases when we are porting, e.g., Groovy code like `str =~ /^[\d\.v]+/`.

I agree that a complete regex library would have the API that separates compilation from matching.

Reply:
(and we should build this library at some point)